extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseQuickpick.ts

```diff
@@ -32,19 +32,26 @@ export async function createNewProjectFromDatabaseWithQuickpick(connectionInfo?:
 	let dbs: string[] | undefined = undefined;
 	while (!dbs) {
 		// Get the list of databases now to validate that the connection is valid and re-prompt them if it isn't
-		try {
-			connectionUri = await vscodeMssqlApi.connect(connectionProfile);
-			dbs = (await vscodeMssqlApi.listDatabases(connectionUri))
-				.filter(db => !constants.systemDbs.includes(db)); // Filter out system dbs
-		} catch (err) {
-			// The mssql extension handles showing the error to the user. Prompt the user
-			// for a new connection and then go and try getting the DBs again
+		let shouldPromptForConnection = connectionProfile?.server === undefined;
+
+		if (!shouldPromptForConnection) {
+			try {
+				connectionUri = await vscodeMssqlApi.connect(connectionProfile);
+				dbs = (await vscodeMssqlApi.listDatabases(connectionUri))
+					.filter(db => !constants.systemDbs.includes(db)); // Filter out system dbs
+			} catch {
+				// The mssql extension handles showing the error to the user.
+				// Prompt the user for a new connection and then go and try getting the DBs again
+				shouldPromptForConnection = true;
+			}
+		}
+
+		if (shouldPromptForConnection) {
```

Comment:
can this be an else so it's more clear that only one of these code blocks gets hit, never both?

Reply:
But it should be, right?  If the attempt to connect/list databases throws, then it still should drop into this prompt for a connection.  At least, that's the existing behavior I preserved; think it makes sense to change it to exit the flow entirely if the connection fails?

Reply:
Ah, if I move the initial set outside the while loop, then it can be `else`.